mtp_api/apps/mtp_auth/serializers.py

```diff
@@ -152,40 +149,63 @@ def validate(self, attrs):
             }
             try:
                 self.initial_data['role'] = managed_roles[role]
+                attrs['role'] = managed_roles[role]
             except KeyError:
                 raise serializers.ValidationError({'role': 'Invalid role: %s' % role})
 
         prisons = self.initial_data.get('prisons')
         if prisons is not None:
-            prison_objects = Prison.objects.filter(
-                nomis_id__in=[prison['nomis_id'] for prison in prisons]
-            )
+            if all([isinstance(prison, Prison) for prison in prisons]):
```

Comment:
so `self.initial_data['prisons']` is sometimes dicts and sometime `Prison`s? how does the latter situation come about?

Reply:
`AccountRequestView.partial_update` loads the prison as part of `get_queryset` https://github.com/ministryofjustice/money-to-prisoners-api/blob/8cf0c9f45584fde3898672584d1ed4b317a206f0/mtp_api/apps/mtp_auth/views.py#L397